### ISSUE TITLE
java.lang.NullPointerException: null values not allowed at org.eclipse.core.internal.registry.ReferenceMap.put(ReferenceMap.java:364)

### ISSUE THREAD

**Issue by col-panic** [eclipse-equinox/p2#117]
I have an ant / download.target.platform job running, as described in https://github.com/elexis/elexis-target/blob/0f81a3ccfbe5026c745670d1c99529de89bf0d5b/target/pom.xml#L381 respectively https://github.com/elexis/elexis-target/blob/2022-06-java17/target/elexis.target.p2mirror.xml.

I'm running the job using Java temurin-17.jdk and Eclipse 2022-06 using `mvn tycho-eclipserun:eclipse-run`. It can be re-rcreated by checking out the `2022-06-java17` branch from https://github.com/elexis/elexis-target/ 

During execution, the following exception is thrown

```
!ENTRY org.eclipse.osgi 4 0 2022-07-22 11:17:58.330
!MESSAGE Application error
!STACK 1
java.lang.NullPointerException: null values not allowed
	at org.eclipse.core.internal.registry.ReferenceMap.put(ReferenceMap.java:364)
	at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.getBundle(RegistryStrategyOSGI.java:169)
	at org.eclipse.core.internal.registry.osgi.RegistryStrategyOSGI.createExecutableExtension(RegistryStrategyOSGI.java:188)
	at org.eclipse.core.internal.registry.ExtensionRegistry.createExecutableExtension(ExtensionRegistry.java:920)
	at org.eclipse.core.internal.registry.ConfigurationElement.createExecutableExtension(ConfigurationElement.java:246)
	at org.eclipse.core.internal.registry.ConfigurationElementHandle.createExecutableExtension(ConfigurationElementHandle.java:63)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:198)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:136)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:402)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:255)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:659)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:596)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1467)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1440)
An error has occurred. See the log file
/Users/marco/git/elexis-target/target/target/eclipserun-work/configuration/1658481477515.log.
```

I don't know what value is being refered to here, or what to do. Any hints appreciated.

**Comment by merks** [comment]
I'm not sure what's all in launched ant application that registers various things, but this purely an OSGi initialization problem where the bundle is null which should never happen:

![image](https://user-images.githubusercontent.com/208716/180446755-ad4a44e9-02d8-4e5e-9928-631fd878a803.png)

Perhaps it would be better to use this without ant:

https://www.eclipse.org/tycho/sitedocs/tycho-extras/tycho-p2-extras-plugin/mirror-mojo.html

**Comment by col-panic** [comment]
Thank you @merks - as far as I remember, I tried without ant in the beginning, but was not capable to use a target file as source, which is where I found https://github.com/eclipse/tycho/issues/176 

I think I tried for some time to get it running, but reverted to the ant solution - as it worked. 

After some more fiddling around, also reverting to older tycho versions, i removed the entire `.m2/repository` and restarted the task. Now it got execute without a problem - the exception did not occur anymore.

Not sure hence what the problem was, but works for me now.
